beacon-chain/state/state-native/setters_payload_header.go

```diff
@@ -21,6 +23,9 @@ func (b *BeaconState) SetLatestExecutionPayloadHeader(val interfaces.ExecutionDa
 
 	switch header := val.Proto().(type) {
 	case *enginev1.ExecutionPayload:
+		if b.version != version.Bellatrix {
```

Comment:
I found it odd and bad that we could have a beacon state that accepted the wrong type of execution payload, so I added that verification for each case and the appropriate test.